src/coreclr/jit/lower.h

```diff
@@ -465,6 +467,37 @@ class Lowering final : public Phase
     // Makes 'childNode' contained in the 'parentNode'
     void MakeSrcContained(GenTree* parentNode, GenTree* childNode) const;
 
+    // Makes 'childNode' regOptional in the 'parentNode'
+    void MakeSrcRegOptional(GenTree* parentNode, GenTree* childNode) const;
+
+    // Tries to make 'childNode' contained or regOptional in the 'parentNode'
+    inline void TryMakeSrcContainedOrRegOptional(GenTree* parentNode, GenTree* childNode) const
```

Comment:
Nit: I think `inline` is not necessary.

Nit: move to `lower.cpp` from the header? `CheckImmedAndMakeContained` is a very similar small function and lives in `lower.cpp`.

Reply:
I generally prefer small helpers like this to be in the header where possible. It helps ensure all the compilers do the "right thing" and don't force method calls.

Reply:
Fixed.